src/lib/jobrunner.rs

```diff
@@ -0,0 +1,409 @@
+use std::{
+    error::Error,
+    fs::{self, remove_file, File},
+    io::{Read, Seek, SeekFrom, Write},
+    os::unix::io::{AsRawFd, RawFd},
+    path::PathBuf,
+    process::{self, Child, Command},
+    sync::Arc,
+    thread,
+};
+
+use lettre::{sendmail::SendmailTransport, EmailAddress, Envelope, SendableEmail, Transport};
+use nix::{
+    fcntl::{fcntl, FcntlArg, OFlag},
+    poll::{poll, PollFd, PollFlags},
+};
+use tempfile::{tempdir, tempfile, NamedTempFile, TempDir};
+use whoami::{hostname, username};
+
+use crate::{queue::QueueJob, Snare};
+
+const READBUF: usize = 8 * 1024; // Bytes. Should be >= PIPE_BUF for performance reasons.
+const WAIT_TIMEOUT: i32 = 1; // Seconds.
```

Comment:
For these timeouts it might be possible to use `Duration::from_secs()` (which is const), then you don't need a comment explaining the unit.

Reply:
I disagree: we'd still need a unit in the comment whatever the case, otherwise the `const` definition doesn't stand on its own.

Reply:
I don't understand. `Duration::from_secs()` tells the reader the unit is seconds.

Reply:
You have to look for where it's used to realise that. By adding it in the comment we make it clearer.

Reply:
Eh? I sense crossed wires.

```
const WAIT_TIMEOUT: Duration = Duration::seconds(1);
```

Reply:
Mea culpa! Yes, I like the idea, so I tried it. Unfortunately `Duration::as_millis()` returns a `u128` when I need an `i32`, so this accidentally made things worse :( Oh well. I've still beefed up the comments in https://github.com/softdevteam/snare/pull/7/commits/e636f4a5b4ff4a18e6085a845fe1a560b1c54744 to get something from the exercise.

Reply:
Ah well. It was worth a shot!